RecoBTag/PixelCluster/plugins/PixelClusterTagInfoProducer.cc

```diff
@@ -0,0 +1,262 @@
+#include "DataFormats/VertexReco/interface/VertexFwd.h"
+
+// For jet
+#include "DataFormats/JetReco/interface/PFJet.h"
+#include "DataFormats/JetReco/interface/PFJetCollection.h"
+
+// For pixel clusters
+#include "DataFormats/SiPixelCluster/interface/SiPixelCluster.h"
+#include "DataFormats/Common/interface/DetSetVectorNew.h"
+#include "DataFormats/Common/interface/Ref.h"
+#include "DataFormats/DetId/interface/DetId.h"
+#include "DataFormats/TrackerCommon/interface/TrackerTopology.h"
+
+// Pixel topology
+#include "DataFormats/SiPixelDetId/interface/PixelSubdetector.h"
+#include "DataFormats/SiPixelDetId/interface/PixelBarrelName.h"
+#include "DataFormats/SiPixelDetId/interface/PixelEndcapName.h"
+#include "DataFormats/SiPixelDetId/interface/PXBDetId.h"
+#include "DataFormats/SiPixelDetId/interface/PXFDetId.h"
+
+// Geometry
+#include "Geometry/Records/interface/IdealGeometryRecord.h"
+#include "Geometry/TrackerGeometryBuilder/interface/PixelGeomDetUnit.h"
+#include "Geometry/TrackerGeometryBuilder/interface/PixelGeomDetType.h"
+#include "Geometry/TrackerGeometryBuilder/interface/TrackerGeometry.h"
+#include "Geometry/Records/interface/TrackerDigiGeometryRecord.h"
+#include "Geometry/CommonDetUnit/interface/GeomDetType.h"
+#include "Geometry/CommonTopologies/interface/PixelTopology.h"
+
+// ROOT
+#include "TVector3.h"
+#include "TLorentzVector.h"
+
+class PixelClusterTagInfoProducer : public edm::global::EDProducer<> {
+public:
+  explicit PixelClusterTagInfoProducer(const edm::ParameterSet&);
+  ~PixelClusterTagInfoProducer() override;
+
+  static void fillDescriptions(edm::ConfigurationDescriptions& descriptions);
+
+private:
+  void produce(edm::StreamID, edm::Event&, const edm::EventSetup&) const override;
+
+  const edm::ParameterSet iConfig;
+
+  const edm::EDGetTokenT<edm::View<reco::Jet> > m_jets;
+  const edm::EDGetTokenT<reco::VertexCollection> m_vertices;
+  const edm::EDGetTokenT<edmNew::DetSetVector<SiPixelCluster> > m_pixelhit;
+  const bool m_isPhase1;
+  const bool m_addFPIX;
+  const int m_minADC;
+  const float m_minJetPt;
+  const float m_maxJetEta;
+  const float m_hadronMass;
+  const int m_nLayers;
+};
+
+PixelClusterTagInfoProducer::PixelClusterTagInfoProducer(const edm::ParameterSet& iConfig)
+    : m_jets(consumes<edm::View<reco::Jet> >(iConfig.getParameter<edm::InputTag>("jets"))),
+      m_vertices(consumes<reco::VertexCollection>(iConfig.getParameter<edm::InputTag>("vertices"))),
+      m_pixelhit(consumes<edmNew::DetSetVector<SiPixelCluster> >(iConfig.getParameter<edm::InputTag>("pixelhit"))),
+      m_isPhase1(iConfig.getParameter<bool>("isPhase1")),
+      m_addFPIX(iConfig.getParameter<bool>("addForward")),
+      m_minADC(iConfig.getParameter<int>("minAdcCount")),
+      m_minJetPt((float)iConfig.getParameter<double>("minJetPtCut")),
+      m_maxJetEta((float)iConfig.getParameter<double>("maxJetEtaCut")),
+      m_hadronMass((float)iConfig.getParameter<double>("hadronMass")),
```

Comment:
were these casts really needed?

Reply:
They were introduced following a comment from @perrotta asking to make these values floats instead of double.

Reply:
but the target values are already floats. The additional explicit cast does nothing.

Reply:
Ok, i see. I removed the casts in the commit i just pushed.